### PR TITLE
[Backend] Get Subscription Charges API Added

### LINKED ISSUE #4: [Backend] Get Subscription Charges API
## Description
Create an API, which will show the subscription charge based on below logic.

From the social stat, take out the max follower - 
For a quarter -> ((80% of the number of followers)*(0.50) / 12) *3  + 20% of ((80% of the number of followers)*(0.50) / 12)*3 
For 6 months -> ((80% of the number of followers)*(0.50) / 12) *6 + 10% of ((80% of the number of followers)*(0.50) / 12)*6 
For a year ->((80% of the number of followers)*(0.50))

Dependent upon - https://github.com/Kajol-Kumari/InfluMart/issues

### PR DESCRIPTION
**Pull request by Hemu21**
## Description

Created the api endpoint for the get subscription plans.

## Issue

[Issue Link](https://github.com/Kajol-Kumari/InfluMart/issues/4) resolve #4 

## Screenshot
sub_plan

### CODE REVIEW

**Comment by Hemu21** [comment]
@Kajol-Kumari  can you please check and merge it

File: backend/controllers/subscriptionController.js [hunk 1]
```diff
@@ -58,4 +73,23 @@ const getSubscription = async (req, res) => {
...
+  if (!platform || !followers) {
+      return res.status(400).json({ error: 'Platform and followers are required' });
+  }
+
+  const charges = calculateCharges(followers);
+
+  return res.json({
+      platform,
+      followers,
+      charges,
```
Comment (Kajol-Kumari) [inline backend/controllers/subscriptionController.js:89]: it should only return the list of subscriptions containing price and duration(quaterly, halfYearly, Annualy)

File: backend/routes/subscriptionRouter.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 const express = require("express");
-const { postSubscription, getSubscription, getPayment } = require("../controllers/subscriptionController");
+const { postSubscription, getSubscription, getPayment, subscriptionPlans } = require("../controllers/subscriptionController");
 const router = express.Router();
 
 
 router.post("/subscription", postSubscription);
 router.get("/subscription/:userName", getSubscription);
 router.get("/payment", getPayment);
+router.post("/plans",subscriptionPlans)
```
Comment (Kajol-Kumari) [inline backend/routes/subscriptionRouter.js:9]: lets make the path as `subscription-plans`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

File: backend/routes/subscriptionRouter.js (hunk 2 above)
Comment (Hemu21) [reply backend/routes/subscriptionRouter.js]: Done.

File: backend/controllers/subscriptionController.js (hunk 1 above)
Comment (Hemu21) [reply backend/controllers/subscriptionController.js]: Ok

**Comment by Hemu21** [comment]
@Kajol-Kumari  Updated can you please check it.

**Review by Kajol-Kumari — APPROVED** [review]
LGTM

**Pull request merged**